examples/routing_service/routing_service_monitoring/c++11/RoutingServiceMonitoring_subscriber.cxx

```diff
@@ -0,0 +1,388 @@
+                                         .mean()
+                              << ", out samples/s (mean): "
+                              << sample.data()
+                                         .value()
+                                         .routing_domain_route()
+                                         .out_samples_per_sec()
+                                         .value()
+                                         .publication_period_metrics()
+                                         .mean()
+                              << std::endl;
+
+                } else if (
+                        sample.data().value()._d()
+                        == RTI::Service::Monitoring::ResourceKind::
+                                ROUTING_SERVICE) {
+                    auto config = getConfig(
+                            configReader,
+                            sample.info().instance_handle());
+                    std::cout << "Periodic data:\n\t > Routing Service "
+                              << config.value()
+                                         .routing_service()
+                                         .application_name()
+                              << "\n\t\t cpu usage (mean): "
+                              << sample.data()
+                                         .value()
+                                         .routing_service()
+                                         .process()
+                                         .value()
+                                         .cpu_usage_percentage()
+                                         .value()
+                                         .publication_period_metrics()
+                                         .mean()
+                              << " %" << std::endl;
+                }
+            }
+        }
+    } catch (const dds::core::Exception &e) {
+        std::cerr << "A Connext exception has been thrown: " << e.what()
+                  << std::endl;
+    } catch (const std::exception &e) {
+        std::cerr << "An std exception has been thrown: " << e.what()
+                  << std::endl;
+    }
+
+    return count;
+}  // The LoanedSamples destructor returns the loan
+
+// Process the samples from the "rti/service/monitoring/event" topic. It shows
+// in the console information related to the Domain Route, Inputs and Outputs
+// (Routing Service).
+int processEventData(
+        dds::sub::DataReader<RTI::RoutingService::Monitoring::Event>
+                &eventReader,
+        dds::sub::DataReader<RTI::RoutingService::Monitoring::Config>
+                &configReader)
+{
+    int count = 0;
+    try {
+        // Take all samples
+        dds::sub::LoanedSamples<RTI::RoutingService::Monitoring::Event>
+                samples = eventReader.take();
+        for (const auto &sample : samples) {
+            if (sample.info().valid()) {
+                count++;
+                if (sample.data().value()._d()
```

Comment:
I had been thinking about this if-else condition around the sample data and I think that we can use a switch-case scenario to improve readability.
What do you think? I didn't go deep in the code and I probably miss something. Please let me know your thoughts :)

Reply:
Thanks for the feedback! I also think that we can use a switch here. In these cases, I usually prefer if/else but that is only a personal preference. Let's say that I am not such a big fan of switch statements, but it is not them, it's me ;)

I will change it!